src/transformStateWithClones.js

```diff
@@ -1,13 +1,55 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} newState
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const returnObjects = [];
+  let container;
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      toClear();
+      container = Object.assign({}, newState);
+      returnObjects.push(container);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      toRemove(actions[i].keysToRemove);
+      container = Object.assign({}, newState);
+      returnObjects.push(container);
+    }
+
+    if (actions[i].type === 'addProperties') {
+      toAdd(actions[i].extraData);
+      container = Object.assign({}, newState);
+      returnObjects.push(container);
+    }
```

Comment:
1. If you used the for loop, it's gonna be better if you set a name for actions[i], for better readability
2. You should use the switch/case statement as written in the checklist
![image](https://user-images.githubusercontent.com/73229127/189705352-1599d44b-5218-4b8a-b119-171c798d964c.png)